### PR TITLE
Add caldav and carddav tests to drone

### PR DESCRIPTION
**Pull request by phil-davis**
## Description
Add caldav-carddav setup and test run steps to drone, add matrix entries.

Rename ``dav/tests/travis`` to ``dav/tests/ci`` so it is agnostic about which CI platform it runs on.

## Motivation and Context
Get more tests on drone.

## How Has This Been Tested?
CI

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Test infrastructure

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by patrickjahns** [comment]
Instead of installing - lets move them into docker containers (as it was done with litmus)

**Comment by DeepDiver1975** [comment]
> Instead of installing - lets move them into docker containers (as it was done with litmus)

can we do this in a second step?

File: apps/dav/appinfo/info.xml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 <info>
 	<id>dav</id>
 	<name>WebDAV</name>
-	<description>ownCloud WebDAV endpoint</description>
+	<description>ownCloud WebDAV endpoint x</description>
```
Comment (DeepDiver1975) [inline apps/dav/appinfo/info.xml]: :question:

**Review by DeepDiver1975 — DISMISSED** [review]

**Comment by patrickjahns** [comment]
> can we do this in a second step? 

Works for me - though I see no time pressure and we can quite easily move this forward the next days

File: apps/dav/appinfo/info.xml (hunk 1 above)
Comment (phil-davis) [reply apps/dav/appinfo/info.xml]: The Travis tests only trigger when something changes in the dav app.
For the moment, I just wanted to see that my rename of folder ``travis`` to ``ci`` did not break the Travis jobs.
I will remove that little commit now that I have seen it work...

And yes, we will remove the Travis jobs. But I guess that will be done in another PR to follow this one.

**Comment by phil-davis** [comment]
This is ready to review again.
Personally I like the "agile" approach - doing small steps that work at each step, e.g. this just extends what was already committed early today for one of the caldav-carddav jobs.
So if someone approves and merges this, then great.

Otherwise let me know and I will ignore this and look at how the whole docker thing would work.

**Comment by phil-davis** [comment]
@DeepDiver1975 @patrickjahns approve-merge here?
Then we can do the next step as another PR.

**Comment by phil-davis** [comment]
@DeepDiver1975 @patrickjahns approve and merge here?
Then we can move on to the next steep.

**Comment by phil-davis** [comment]
Ping @patrickjahns @DeepDiver1975 
I rebased just now, to run CI again and confirm it will work OK still.
OK to approve and merge this, and then move on to another iteration in the future?

**Comment by phil-davis** [comment]
Ping @patrickjahns @DeepDiver1975 
I rebased again just now, to resolve conflictsl.
OK to approve and merge this, and then move on to another iteration in the future?

**Comment by DeepDiver1975** [comment]
the tests don't seem to run on travis anymore -- but as far as I can tell this is unrelated to this change.

**Comment by DeepDiver1975** [comment]
for the drone execution we better *one day* pipe the php builtin web server output to a file on it's own.
same applies to the acceptance tests

**Comment by phil-davis** [comment]
I think the Travis thing is because there are some smarts in the scripts - if nothing changes in the ``apps/dav`` folder then it does not bother actually running any tests:
Travis log has ``Not testing DAV``

**Comment by DeepDiver1975** [comment]
ah - see 

https://github.com/owncloud/core/blob/a3edfb87e8fd3d4f359c1ac3a10b70c205fdea18/.travis.yml#L15

**Comment by DeepDiver1975** [comment]
tests are only executed if code within the dav app was changed -> got to go

**Review by DeepDiver1975 — APPROVED** [review]

**Comment by phil-davis** [comment]
Backport ``stable10`` #31790

**Pull request merged**